pkg/controller/rollout/rollout_controller.go

```diff
@@ -79,6 +96,18 @@ func (r *RolloutReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		return ctrl.Result{}, err
 	}
 
+	modified, err := util.AddDynamicWatcher(runtimeController, workloadHandler, watchedWorkload, rollout.Spec.ObjectRef.WorkloadRef)
```

Comment:
Do not pass in `watchedWorkload`, you should check if the GVK is watched here. If not, then add the watcher dynamically.